### PR TITLE
Pull in fireplace's test commands and helpers

### PR DESCRIPTION
**Pull request by mstriemer**
This shares the testing commands across projects similar to how the serve, package, etc commands are shared.

It also shares the casper shim that is needed to run under slimerJS and what seems to be globally usable helpers for casper tests.

The shim will now load `marketplace-gulp/casper-helpers.js` and merge `tests/lib/helpers.js` into that object to create the `helpers` object. `appList` and `constants` are no longer globally available in fireplace.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,3 +39,37 @@ init:
...
 update:
 	@echo "'make update' has been removed. Use 'make install' instead."
+
+test:
+	make lint && make unittest && make uitest
+
+uitest:
+	make uitest-phantom && make uitest-slimer
+
+uitest-phantom: css templates
```
Comment (ngokevin) [inline Makefile]: On a local branch, I removed the ```css templates``` dependency directives. Starting up the webserver will do this so we shouldn't need to do it all the time. It takes about 2-3 seconds to do each time so it slows down testing.

**Comment by ngokevin** [comment]
Can you put more info in the PR description of the overall plan of consuming?

File: tasks.js [hunk 2]
```diff
@@ -77,6 +80,46 @@ gulp.task('clean', function() {
 });
 
 
+function uploadCaptures(options) {
```
Comment (ngokevin) [inline tasks.js]: i'd probably put this in a tests.js task file

File: casper-helpers.js [hunk 3]
```diff
@@ -0,0 +1,323 @@
+var system = require('system');
```
Comment (ngokevin) [inline casper-helpers.js]: Not sure if this belongs here but it's a good step. I'd put this and the ```casper-shim``` in a ```marketplace-gulp/tests``` folder

**Comment by ngokevin** [comment]
r+

**Comment by mstriemer** [comment]
I updated the description. To load `appList` you need to call `var appList = localRequire('tests/lib/app_list')`. That's fine but if we're going to load helpers like that maybe we should add something like `var appList = helpers.load('app_list')`? I had a few crazy ideas of how to make this easier but it might not be worth optimizing.

**Comment by ngokevin** [comment]
nice, love it.

**Comment by mstriemer** [comment]
I added `helpers.load` and `helpers.npm`. The initial requires in `tests/ui/app_list.js` now look like:

```js
var jsuri = helpers.npm('jsuri/Uri');
var appList = helpers.load('app_list');
var constants = helpers.load('constants');
```

**Comment by ngokevin** [comment]
awesome :+1:

**Pull request closed without merging**